cmd/kubeadm/app/cmd/reset.go

```diff
@@ -183,10 +184,15 @@ func (r *Reset) Run(out io.Writer) error {
 	return nil
 }
 
-func reset(execer utilsexec.Interface, dockerCheck preflight.Checker, criSocketPath string) {
-	crictlPath, err := execer.LookPath("crictl")
-	if err == nil {
-		resetWithCrictl(execer, dockerCheck, criSocketPath, crictlPath)
+func reset(execer utilsexec.Interface, dockerCheck preflight.Checker, criCheck preflight.Checker, criSocketPath string) {
+	if criSocketPath != kubeadmapiv1alpha2.DefaultCRISocket {
```

Comment:
Having a lot of detection for docker vs. crictrl spread across source code is not good in my opinion.
It will be a lot better to refactor this code a bit, move all detection to helper function in `preflight/checks.go` and here keep only simple code resetWithDocker/resetWithCrictl based on output of helper function.

Reply:
Agree.